MDANSE/Src/MDANSE/Trajectory/MdanseTrajectory.py

```diff
@@ -490,111 +478,85 @@ def has_variable(self, variable: str) -> bool:
-        global atom database.
-
-        Parameters
-        ----------
-        symbol : str
-            Atom type label
-        property_name : str
-            Name of the atom property
-
-        Returns
-        -------
-        Union[float, int, str]
-            The value from the database converted to its type
-
-        Raises
-        ------
-        KeyError
-            The property name is not in the trajectory's database
-
-        """
+    def get_atom_property(self, symbol: str, property: str):
         if not self._has_database:
-            return ATOMS_DATABASE.get_atom_property(symbol, property_name)
-        if symbol not in self._has_atoms:
+            return ATOMS_DATABASE.get_atom_property(symbol, property)
+        elif symbol not in self._has_atoms:
             if symbol in self._h5_file["/atom_database"]:
                 self._has_atoms.append(symbol)
             else:
-                return ATOMS_DATABASE.get_atom_property(symbol, property_name)
-        if property_name not in self._property_map:
+                return ATOMS_DATABASE.get_atom_property(symbol, property)
+        if property not in self._property_map:
             temp = np.where(
                 self._h5_file["/atom_database/property_labels"][:]
-                == property_name.encode("utf-8"),
+                == property.encode("utf-8")
             )[0]
             if len(temp) == 0:
-                if property_name == "dummy":
+                if property == "dummy":
                     try:
-                        return ATOMS_DATABASE.get_atom_property(symbol, property_name)
+                        return ATOMS_DATABASE.get_atom_property(symbol, property)
                     except KeyError:
                         if (
                             "_" in symbol
                         ):  # this is most likely an artificial atom from a molecule
                             return 0  # the molecule atoms are not dummy
                 else:
                     raise KeyError(
-                        f"Property {property_name} is not in the trajectory's internal database."
+                        f"Property {property} is not in the trajectory's internal database."
                     )
             index = temp.flatten()[0]
-            self._property_map[property_name] = index
-        index = self._property_map[property_name]
+            self._property_map[property] = index
+        index = self._property_map[property]
         if index not in self._data_types:
             self._data_types[index] = self._h5_file["/atom_database/property_types"][
                 index
             ]
         data_type = self._data_types[index]
+        if index not in self._data_units:
+            data_unit = "none"
+            try:
+                unit_lookup = self._h5_file["/atom_database/property_units"]
+            except KeyError:
+                if not self.warned_about_complex_numbers:
+                    LOG.warning(
```

Comment:
`warnings.warn` has this behaviour by default (unless you set `warnings.simplefilter('always')` and should be compatible with `LOG`.